app/src/org/commcare/dalvik/activities/EntityMapActivity.java

```diff
@@ -1,288 +1,186 @@
-                                gp = new GeoPoint(lat, lng);
-                            }
-                        } catch(Exception ex) {
-                            //We might not have a geopoint at all. Don't even trip
-                        }
-                        
-                        boolean cached = false;
-                        try {
-                            GeocodeCacheModel record = geoCache.getRecordForValue(GeocodeCacheModel.META_LOCATION, val);
-                            cached = true;
-                            if(record.dataExists()){
-                                gp = record.getGeoPoint();
-                            }
-                        } catch(NoSuchElementException nsee) {
-                            //no record!
-                        }
-                        
-                        //If we don't have a geopoint, let's try to find our address
-                        if (!cached && location != null) {
-                            try {
-                                List<Address> addresses = mGeoCoder.getFromLocationName(val, 3, boundHints[0], boundHints[1], boundHints[2], boundHints[3]);
-                                for(Address a : addresses) {
-                                    if(a.hasLatitude() && a.hasLongitude()) {
-                                        int lat = (int) (a.getLatitude() * 1E6);
-                                        int lng = (int) (a.getLongitude() * 1E6);
-                                        gp = new GeoPoint(lat, lng);
-                                        
-                                        geoCache.write(new GeocodeCacheModel(val, lat, lng));
-                                        legit++;
-                                        break;
-                                    }
-                                }
-                                
-                                //We didn't find an address, make a miss record
-                                if(gp == null) {
-                                    geoCache.write(GeocodeCacheModel.NoHitRecord(val));
-                                }
-                            } catch (StorageFullException | IOException e1) {
-                                e1.printStackTrace();
-                            }
-                        }
-                        
-                        //Ok, so now we have an address or not. If we _do_ have one, let's have some fun
-                        
-                        if(gp != null) {
-                            OverlayItem overlayItem = overlayFactory.generateOverlay(gp, e);
-                            mEntityOverlay.addOverlay(overlayItem, e.getElement());
-                        }
-                        else { 
-                            bogus++;
+                    String address = entity.getFieldString(i).trim();
+                    if(address != null && !"".equals(address)) {

```

Comment:
you don't need to check for null after calling `trim()`, since that will raise a null pointer exception and the null check will never be reached.
Looking at the implementation of `getFieldString`, it seems to always return a String instance, so you don't need the check in general